pyccel/ast/core.py

```diff
@@ -210,11 +210,29 @@ def broadcast(shape_1, shape_2):
             errors.report(msg,severity='fatal')
     return tuple(new_shape)
 
+def handle_precedence(args, my_precedence, commutative = False):
+    precedence = [getattr(a, 'precedence', 17) for a in args]
+
+    if min(precedence) <= my_precedence:
+
+        new_args = []
+
+        for a,p in zip(args, precedence):
+            if (p < my_precedence or (not commutative and p == my_precedence)):
+                new_args.append(PyccelAssociativeParenthesis(a))
+            else:
+                new_args.append(a)
+        args = tuple(new_args)
+
+    return args
+
 class PyccelOperator(Expr, PyccelAstNode):
 
     def __init__(self, *args):
 
         if self.stage == 'syntactic':
+            args = handle_precedence(args, self.precedence, self._commutative)
+            self._args = tuple(args)
```

Comment:
Doesn't `handle_precedence` return a `tuple` object?

Reply:
You are right. Oops